src/components/calendar/bl-calendar.ts

```diff
@@ -0,0 +1,545 @@
+import { CSSResultGroup, html, LitElement } from "lit";
+import { customElement, property, state } from "lit/decorators.js";
+import { classMap } from "lit/directives/class-map.js";
+import { event, EventDispatcher } from "../../utilities/event";
+import "../button/bl-button";
+import "../icon/bl-icon";
+import {
+  CALENDAR_TYPES,
+  CALENDAR_VIEWS,
+  FIRST_MONTH_INDEX,
+  LAST_MONTH_INDEX,
+} from "./bl-calendar.constant";
+import style from "./bl-calendar.css";
+import {
+  Calendar,
+  CalendarDate,
+  CalendarType,
+  CalendarView,
+  CalendarDay,
+  DayValues,
+  RangePickerDates,
+} from "./bl-calendar.types";
+
+/**
+ * @tag bl-calendar
+ * @summary Baklava Calendar component
+ **/
+
+@customElement("bl-calendar")
+export default class BlCalendar extends LitElement {
+  /**
+   *Defines the calendar types, available types are single,multiple and range
+   */
+  @property()
+  type: CalendarType = CALENDAR_TYPES.single;
+
+  /**
+   *Defines the minimum date value for the calendar
+   */
+  @property({ type: Date, attribute: "min-date", reflect: true })
+  minDate: Date;
+
+  /**
+   * Defines the maximum date value for the calendar
+   */
+  @property({ type: Date, attribute: "max-date", reflect: true })
+  maxDate: Date;
+
+  /**
+   * Defines the start day of the calendar (1 defines monday)
+   */
+  @property({ type: Number, attribute: "start-of-week", reflect: true })
+  startOfWeek: DayValues = 0;
+
+  /**
+   * Defines the unselectable dates for calendar
+   */
+  @property({ type: Array, attribute: "disabled-dates", reflect: true })
+  disabledDates: Date[];
+
+  /**
+   * Defines the calendar language
+   */
+  @property()
+  locale: string = document.documentElement.lang;
+
+  @state()
+  private _selectedDates: CalendarDate[] = [];
+
+  @state()
+  private _selectedRangeDates: RangePickerDates = { startDate: undefined, endDate: undefined };
+
+  @state()
+  private _calendarMonth: number = new Date().getMonth();
```

Comment:
Instead of defining `new Date()` for `_calendarMonth` and `_calendarYear` we can define it once and use.